File: python/excercise739/daily_temperatures.py
from typing import List


def daily_temperatures(temperatures: List[int]) -> List[int]:
    n = len(temperatures)
    ans = [0] * n

    stk = []
    for i in range(n):
        while len(stk) > 0 and temperatures[stk[-1]] < temperatures[i]:
            ans[stk[-1]] = i - stk[-1]
            stk.pop()
        stk.append(i)

    return ans

File: go/excercise739/daily_temperatures.go
package daily_temperatures

func dailyTemperatures(temperatures []int) []int {
	n := len(temperatures)
	ans := make([]int, n)

	stk := make([]int, 0)
	for i := 0; i < n; i++ {
		for len(stk) > 0 && temperatures[stk[len(stk)-1]] < temperatures[i] {
			ans[stk[len(stk)-1]] = i - stk[len(stk)-1]
			stk = stk[:len(stk)-1]
		}
		stk = append(stk, i)
	}

	return ans
}
